src/interpreter.rs

```diff
@@ -4,12 +4,12 @@ use std::collections::HashMap;
 type Stack<T> = Vec<T>;
 
 #[derive(Default)]
-pub struct Interpreter<'a, T> {
+pub struct Interpreter<'a, 'b, T: 'b> {
```

Comment:
Did that take part in the main problem? I think I monkeyed this part until the compiler shut up :speak_no_evil: